testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergProcedureCalls.java

```diff
@@ -56,6 +57,68 @@ public void testMigrateHiveTable()
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " (x timestamp(3)) WITH (format='ORC')");
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 10:11:12.123'");
+
+        onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')");
+
+        assertThat(onTrino().executeQuery("SELECT x FROM " + icebergTableName))
+                .containsOnly(row(Timestamp.valueOf("2021-01-01 10:11:12.123")));
+        onSpark().executeQuery("SET `spark.sql.iceberg.handle-timestamp-without-timezone` = TRUE");
+        assertThat(onSpark().executeQuery("SELECT x FROM " + sparkTableName))
+                .containsOnly(row(Timestamp.valueOf("2021-01-01 10:11:12.123")));
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + icebergTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testMigrateTimestampMillisHiveTableWithAvro()
+    {
+        String tableName = "test_migrate_timestamp" + randomNameSuffix();
+        String hiveTableName = "hive.default." + tableName;
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " (x timestamp(3)) WITH (format='AVRO')");
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 10:11:12.123'");
+
+        assertThatThrownBy(() -> onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')"))
+                .hasMessageContaining("Migrating timestamp(3) type with AVRO format is not supported.");
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testMigrateTimestampMillisHiveTableWithParquet()
+    {
+        String tableName = "test_migrate_timestampPar_" + randomNameSuffix();
+        String hiveTableName = "hive.default." + tableName;
+        String icebergTableName = "iceberg.default." + tableName;
+        String sparkTableName = "iceberg_test.default." + tableName;
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " (x timestamp(3)) WITH (format='PARQUET')");
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 10:11:12.123'");
+
+        onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')");
+        assertThat(onTrino().executeQuery("SELECT * FROM " + icebergTableName))
+                .containsOnly(row(Timestamp.valueOf("2021-01-01 04:26:12.123"))); // Test run within TZ Asia/Kathmandu UTC+5:45
+        onSpark().executeQuery("set `spark.sql.iceberg.handle-timestamp-without-timezone` = TRUE");
+        onSpark().executeQuery("ALTER TABLE " + sparkTableName + " SET TBLPROPERTIES ('read.parquet.vectorization.enabled'='false')");
```

Comment:
why? document

Reply:
Our version of Spark can not read int96 timestamp  in imported data with vectorized reader so old batch reader needs to be used.
https://github.com/apache/iceberg/pull/6962

Reply:
Sorry for being to brief. When saying "document" i usually mean to capture the reasoning in the code, e.g. as a code comment.